### ISSUE TITLE
Error: key encryption algorithm not supported

### ISSUE THREAD

**Issue by aw1875** [node-saml/passport-saml#618]
I am trying to use this package to help facilitate authentication with my school's shibboleth login. It seems like I have set everything up correctly but I am getting this error returned after I get sent to my callback route:

```
Error: key encryption algorithm http://www.w3.org/2009/xmlenc11#rsa-oaep not supported
    at decryptKeyInfo (/var/www/shibb/node_modules/xml-encryption/lib/xmlenc.js:246:13)
    at Object.decrypt (/var/www/shibb/node_modules/xml-encryption/lib/xmlenc.js:182:24)
    at internal/util.js:297:30
    at new Promise (<anonymous>)
    at Object.decrypt (internal/util.js:296:12)
    at Object.decryptXml (/var/www/shibb/node_modules/passport-saml/lib/node-saml/xml.js:39:93)
    at SAML.validatePostResponseAsync (/var/www/shibb/node_modules/passport-saml/lib/node-saml/saml.js:591:50)
```

I've tried messing with the signing using `signatureAlgorithm` and nothing seems to work. I'm not entirely sure if this is an issue related to passport-saml or something that is setup incorrectly with my certificates from my idp (or even an issue on my idp's side). If anyone has come across this issue or knows how to fix it I would really appreciate the help!

**Comment by cjbarth** [comment]
`oaep` encryption isn't supported, just as the error message says; you need to use a supported encryption algorithm, which should match what your IdP uses. See https://github.com/node-saml/passport-saml/blob/f1e00b64c21a725f545e675cd810bbaa435a3972/src/node-saml/saml.ts#L1403

**Comment by aw1875** [comment]
@cjbarth I understand that `oaep` isn't supported but wasn't 100% sure so appreciate that! Is this related to decrypting the response returned from my IdP? Also, I see the file you attached shows me what is supported, is there some way to specify using one of these methods instead or did I just miss that in the documentation?

**Comment by cjbarth** [comment]
Your IdP should ingest the metadata from `passport-saml` and only use supported encryption. That list I referenced is in the metadata.

**Comment by aw1875** [comment]
Thank you! I have reached out to my IdP to make sure this can be sorted out appropriately. Appreciate your help!

**Comment by aw1875** [comment]
@cjbarth Hi sorry another quick question when you have a moment. I reached out to my IdP and they have modified the metadata so that issue is fixed. I am now getting another issue and just wanted to double check if I have an understanding of the reason why. I am now getting:

```
Error: Invalid signature from encrypted assertion
    at SAML.validatePostResponseAsync (/var/www/shibb/node_modules/passport-saml/lib/node-saml/saml.js:598:27)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

I noticed that my SSL certificate has expired so I figured it was due to this issue. Am I correct in assuming this or is this some sort of other issue?

**Comment by aw1875** [comment]
Hi @cjbarth my IdP has modified the metadata to solve the initial issue. However, I now have this issue whenever I am sent to my callback route:

```
Error: Invalid signature from encrypted assertion
    at SAML.validatePostResponseAsync (/var/www/shibb/node_modules/passport-saml/lib/node-saml/saml.js:598:27)
```

I've found a few sites talking about this issue but they are all related to the cert. I have checked the cert and compared it with the metadata and it is all accurate so I'm not entirely sure what the issue is.
